src/blockchain/node/wallet/subchain/statemachine/Scan.cpp

```diff
@@ -232,7 +232,17 @@ auto Scan::Imp::work() noexcept -> bool
     }
 
     const auto height = current().first;
-    const auto rescan = parent_.rescan_progress_.load();
+    const auto rescan = [&]() -> block::Height {
```

Comment:
Code in lines 235...245 could happily be the following 2 lines:
auto handle = parent_.progress_position_.lock();
auto rescan = handle->has_value() ? handle->value().first : -1;